packages/dx-grid-core/src/plugins/paging-state/computeds.js

```diff
@@ -1,8 +1,16 @@
-export const paginate = (rows, pageSize, page) =>
-  rows.slice(pageSize * page, pageSize * (page + 1));
+const ALL_PAGES_TITLE = 'All';
+
+export const paginate = (rows, pageSize, page) => (
+  pageSize ?
+    rows.slice(pageSize * page, pageSize * (page + 1)) :
+    rows
+);
 
 export const ensurePageHeaders = (rows, pageSize) => {
   const result = rows.slice();
+  if (!pageSize) {
```

Comment:
Yes, but this copy is needed only for processing listed below in this function. We can return original rows if processing is not needed.